rbs/MethodTypeTranslator.cc

```diff
@@ -0,0 +1,307 @@
+#include "MethodTypeTranslator.h"
+#include "TypeTranslator.h"
+#include "absl/strings/escaping.h"
+#include "ast/AttrHelper.h"
+#include "ast/Helpers.h"
+#include "core/GlobalState.h"
+#include "core/errors/rewriter.h"
+
+using namespace sorbet::ast;
+
+namespace sorbet::rbs {
+
+namespace {
+
+core::NameRef expressionName(core::MutableContext ctx, const ast::ExpressionPtr *expr) {
+    core::NameRef name;
+
+    typecase(
+        *expr, [&](const ast::UnresolvedIdent &p) { name = p.name; },
+        [&](const ast::OptionalArg &p) { name = expressionName(ctx, &p.expr); },
+        [&](const ast::RestArg &p) { name = expressionName(ctx, &p.expr); },
+        [&](const ast::KeywordArg &p) { name = expressionName(ctx, &p.expr); },
+        [&](const ast::BlockArg &p) { name = expressionName(ctx, &p.expr); },
+        [&](const ast::ExpressionPtr &p) {
+            if (auto e = ctx.beginError(expr->loc(), core::errors::Rewriter::RBSError)) {
+                e.setHeader("Unexpected expression type: {}", p.showRaw(ctx));
+            }
+
+            name = ctx.state.enterNameUTF8("<error>");
+        });
+
+    return name;
+}
+
+struct RBSArg {
+    core::LocOffsets loc;
+    rbs_ast_symbol_t *name;
+    rbs_node_t *type;
+    bool optional;
```

Comment:
Given questions elsewhere about `optional`, maybe this should be an `enum class { Required, Optional }`?

Reply:
We actually don't need it anymore, I was using it to check that the optional parameters stated in the RBS signature were matching the parameters having a default value in the method definition but we can delegate this check to further down the pipeline.

Reply:
That makes a lot of sense!